mlir/lib/Interfaces/ViewLikeInterface.cpp

```diff
@@ -114,7 +114,8 @@ void mlir::printDynamicIndexList(OpAsmPrinter &printer, Operation *op,
                                  OperandRange values,
                                  ArrayRef<int64_t> integers,
                                  ArrayRef<bool> scalables, TypeRange valueTypes,
-                                 AsmParser::Delimiter delimiter) {
+                                 AsmParser::Delimiter delimiter,
+                                 bool hasSameTypeDynamicValues) {
   char leftDelimiter = getLeftDelimiter(delimiter);
```

Comment:
In case we keep the `valueTypes` parameter, I think there should be an `assert` that checks the number of elements in `valueTypes`.

Reply:
Also, can we assert that `TypeRange(values) == valueTypes`?